clang/lib/CodeGen/CGLoopInfo.cpp

```diff
@@ -706,8 +706,15 @@ void LoopInfoStack::push(BasicBlock *Header, clang::ASTContext &Ctx,
     // 1 - disable unroll.
     // other positive integer n - unroll by n.
     if (OpenCLHint || UnrollHint) {
-      ValueInt = OpenCLHint ? OpenCLHint->getUnrollHint()
-                            : UnrollHint->getUnrollHint();
+      if (OpenCLHint)
```

Comment:
More like: 

    ValueInt = 0; // Is not not already?
    if (OpenCLHint)
        ValueInt = OpenCLHint->getUnrollHint();
    else if (Expr *VH = UnrollHint->getUnrollHintExpr())
        ValueInt = VH->EvaluateKnownConstInt(Ctx).getSExtValue();

Reply:
ValueInt is 1 by default, so it's more like:

    if (OpenCLHint)
        ValueInt = OpenCLHint->getUnrollHint();
    else if (Expr *VH = UnrollHint->getUnrollHintExpr())
        ValueInt = VH->EvaluateKnownConstInt(Ctx).getSExtValue();
    else
        ValueInt = 0;

Reply:
Right, so instead of the 'else' you can just do the assignment first (like I did there).